pkg/registry/extensions/daemonset/strategy.go

```diff
@@ -70,6 +70,15 @@ func (daemonSetStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, ol
 	// update is not allowed to set TemplateGeneration
 	newDaemonSet.Spec.TemplateGeneration = oldDaemonSet.Spec.TemplateGeneration
 
+	// Update is not allowed to set Spec.Selector for all groups/versions except extensions/v1beta1.
+	// If RequestInfo is nil, it is better to revert to old behavior (i.e. allow update to set Spec.Selector)
+	// to prevent unintentionally breaking users who may rely on the old behavior.
+	// TODO(#50791): after v1beta1 is retired, move selector immutability check to validation codes
+	requestInfo, found := genericapirequest.RequestInfoFrom(ctx)
+	if found && !(requestInfo.APIGroup == "extensions" && requestInfo.APIVersion == "v1beta1") {
+		newDaemonSet.Spec.Selector = oldDaemonSet.Spec.Selector
```

Comment:
>I suppose we already do similar things for fields like TemplateGeneration. Have you checked to see what tools like kubectl apply do if asked to apply a change to this field? I assume they keep re-sending a patch thinking they can update the value

Yes, `kubectl apply` will send the patch, but the update does not happen. We have to inform users about this change. Release note helps, but i think we should do more.